apis/kahu/v1beta1/backup.go

```diff
@@ -0,0 +1,159 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+type Backup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BackupSpec   `json:"spec,omitempty"`
+	Status BackupStatus `json:"status,omitempty"`
+}
+
+type BackupSpec struct {
+	// +optional
+	ReclaimPolicy ReclaimPolicyType `json:"reclaimPolicy,omitempty"`
+
+	// +nullable
+	// +optional
+	MetadataLocation *BackupLocation `json:"metadataLocation,omitempty"`
+
+	// +optional
+	PreExecHook ResourceHookSpec `json:"preExecHook,omitempty"`
+
+	// +optional
+	PostExecHook ResourceHookSpec `json:"postExecHook,omitempty"`
```

Comment:
The current release only handles Metadata backup and restore. IMO, Attributes such as ```PreExecHook``` and ```PostExecHook``` can be dumped currently